release/releaser.go

```diff
@@ -261,8 +261,44 @@ func selectServices(rc *ReleaseContext, spec *flux.ReleaseSpec, results flux.Rel
 	}
 
 	// Build list of filters
-	var filtList []ServiceFilter
+	filtList, err := filters(spec, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// For backwards-compatibility, there's two fields: ServiceSpec
+	// and ServiceSpecs. An entry in ServiceSpec takes precedence.
+	switch spec.ServiceSpec {
```

Comment:
I removed the legacy `ServiceSpec` from `ReleaseSpec` since it wasn't used anymore. This simplified the code dramatically. (All methods now pass a `[]ServiceSpec`)